#! /bin/sh -e

# DP: Declare a data_start symbol in crt1.o

if [ $# -ne 2 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch -d "$2" -f --no-backup-if-mismatch -p1 < $0;;
    -unpatch) patch -d "$2" -f --no-backup-if-mismatch -R -p1 < $0;;
    *)
	echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
	exit 1
esac
exit 0

--- glibc-2.2.5/sysdeps/hppa/elf/start.S~	Fri Feb  8 19:29:36 2002
+++ glibc-2.2.5/sysdeps/hppa/elf/start.S	Fri Feb  8 19:29:59 2002
@@ -56,3 +56,10 @@
 
 	.procend
 
+/* Define a symbol for the first piece of initialized data.  */
+	.data
+	.globl __data_start
+__data_start:
+	.long 0
+	.weak data_start
+	data_start = __data_start


